In [1]:
#Import necessary modules, libraries
import tensorflow as tf
from tensorflow import keras
from tensorflow.keras import layers

In [2]:
#Define a function named build_model for FFNN model building
def build_model():
    # input layer with 28x28 image
    inputs = Input((28, 28))
    # flattening the input layer to make 1D representation of the 2D image(28x28)
    x = Flatten()(inputs)
    # a dense layer with 32 neuron and sigmoid activation function
    x = Dense(32, activation = 'sigmoid')(x)
    # another dense layer with 16 neuron and sigmoid activation function
    x = Dense(16, activation = 'sigmoid')(x)
    # output layer with 3 neuron, here 3 also represents the class in output layer
    outputs = Dense(3)(x)
    # a model; supplied the inputs and outputs
    model = Model(inputs, outputs)
    # summary describe the model as it's layer, output shape, parameter and more
    model.summary()
    # this function determines the loss measured with means squared error(mse)
    model.compile(loss = 'mse')
    # build_model function will return the model
    return model  
    

inputs = Input((28, 28))
x = Flatten()(inputs)
x = Dense(32, activation = 'sigmoid')(x)
x = Dense(16, activation = 'sigmoid')(x)
outputs = Dense(3)(x)
model = Model(inputs, outputs)
#model.summary()
model.compile(loss = 'mse')

In [4]:
#build_model().layers

In [32]:
# define the main function
def main():
    # in this main funtion call the build_model function
    model = build_model()

In [33]:
# Execute the main function
if __name__ == "__main__":
	main()

Model: "model_9"
_________________________________________________________________
 Layer (type)                Output Shape              Param #   
 input_10 (InputLayer)       [(None, 28, 28)]          0         
                                                                 
 flatten_9 (Flatten)         (None, 784)               0         
                                                                 
 dense_33 (Dense)            (None, 32)                25120     
                                                                 
 dense_34 (Dense)            (None, 16)                528       
                                                                 
 dense_35 (Dense)            (None, 3)                 51        
                                                                 
Total params: 25,699
Trainable params: 25,699
Non-trainable params: 0
_________________________________________________________________


The number of parameters is 25120 because with every hidden unit you have 784 input weights and 1 weight of connection with bias. This means that every hidden unit gives you (784+1=)785 parameters. You have 32 units so it sums up to 785*32=25120.

Similarly, (32+1)*16=528
(16+1)*3=51